File: frontend/src/types/chatbot.ts
export type ActionType =
  | 'static_reply'
  | 'collect_input'
  | 'knowledge_base'
  | 'api_call'
  | 'llm_generate'
  | 'end_conversation';

export interface Action {
  type: ActionType;
  message?: string; // static_reply, end_conversation
  variable?: string; // collect_input
  prompt?: string; // collect_input
  kb_id?: number; // knowledge_base
  endpoint?: string; // api_call
  method?: string; // api_call
}

export interface WorkflowNode {
  id: string;
  intent?: string;
  action: Action;
  next_node?: string;
  reply_template?: string;
  position: { x: number; y: number };
}

export interface Workflow {
  nodes?: WorkflowNode[];
}

export interface ChatBot {
  id: number;
  user_id: number;
  name: string;
  description: string | null;
  api_key: string;
  workflow: string | null;
  config: string | null;
  created_at: string;
  updated_at: string;
}

File: backend/models/chatbot.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type ChatBot struct {
	ID          uint           `gorm:"primaryKey" json:"id"`
	UserID      uint           `gorm:"not null;index" json:"user_id"`
	Name        string         `gorm:"not null" json:"name"`
	Description *string        `json:"description"`
	APIKey       string         `gorm:"uniqueIndex" json:"api_key"`
	Workflow     *string        `gorm:"type:jsonb" json:"workflow"`
	Config       *string        `gorm:"type:jsonb" json:"config"`
	Integrations *string        `gorm:"type:jsonb" json:"integrations"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}
